### PR TITLE
Change CAISOClient.get_ancillary_services to return list of dicts, bu…

### PR DESCRIPTION
**Pull request by ndavis6**
…mp version

### CODE REVIEW

File: pyiso/caiso.py [hunk 1]
```diff
@@ -486,7 +497,7 @@ def fetch_oasis(self, payload={}, return_all_files=False):
             return default_return_val
 
         # load xml into soup
-        soup = BeautifulSoup(content[0])  # Check the first file for errors
+        soup = BeautifulSoup(content[0], 'lxml')  # Check the first file for errors
```
Comment (ndavis6) [inline pyiso/caiso.py]: This will cause BeautifulSoup to throw an error if lxml is not installed rather than use a parser with potentially different behavior.  Is this desired behavior?  My thought was that we would rather get a clear error, than potentially an error somewhere else in processing that was dependent on the machine configuration.

File: pyiso/caiso.py [hunk 2]
```diff
@@ -283,7 +284,24 @@ def get_AS_dataframe(self, anc_region, latest=True, start_at=False, end_at=False
         MW columns holds $/MW float.
         If no data, returns an empty dataframe.
         """
-        # set args
+        # set args, handle latest differently than LMP b/c daily publication schedule
```
Comment (aschn) [inline pyiso/caiso.py]: Could this be made a separate method with its own tests? That would help me understand what edge cases it's intending to handle. Also, what if we make latest=True return today always, and tomorrow=True return tomorrow if it's available yet... would that work for the downstream problem you're addressing here?

Comment (ndavis6) [inline pyiso/caiso.py]: The problem is that handle-options takes latest=true to mean a 20-minute window.  AS prices are done hourly, so it returns nothing (except maybe if you are calling near the hour).  CAISO publishes the DAM AS prices at 1:00PM the day before the trading day.  This code gets tomorrows AS prices if it is after 1PM and todays if it is before 1PM.  latest=True returns today always also works.  Users can always set start_at / end_at if they want something else

Comment (aschn) [inline pyiso/caiso.py]: Ah, the 20-min window problem makes sense. But having latest=True return a full day of data, especially a full day of data in the future, is super different from the behavior of latest=True on all other methods. Just trying to think of the simplest way to keep latest=True consistent while also useful...

Comment (ndavis6) [inline pyiso/caiso.py]: It could return a 1-hour (+ 1 second) window.  That would guarantee some return and might be more consistent with the other functions.

Comment (aschn) [inline pyiso/caiso.py]: That sounds perfect to me!

Comment (ndavis6) [inline pyiso/caiso.py]: Updated
